specs/kyber/src/serialize.rs

```diff
@@ -0,0 +1,108 @@
+use crate::ring::RingElement;
+use crate::field::FieldElement;
+use crate::bit_vector::*;
+use crate::parameters;
+
+///
+/// Given a series of bytes representing a ring element in |ring_element_bytes|,
+/// first convert them into a vector of bits in little-endian order; i.e.
+/// the least significant |BITS_PER_COEFFICIENT| of |ring_element_bytes[0]|
+/// are the first set of bits in the bitstream.
+
+/// This vector is deserialized into a RingElement structure. The first
+/// |BITS_PER_COEFFICIENT| represent the first coefficient of the ring element,
+/// the second |BITS_PER_COEFFICIENT| the second coefficient, and so on.
+///
+/// This function implements Algorithm 3 of the Kyber Round 3 specification,
+/// which is reproduced below:
+///
+/// ```plaintext
+/// Input: Byte array B ∈ B32l
+/// Output: Polynomial f ∈ Rq
+/// (β0, . . . , β256l−1) := BytesToBits(B)
+/// for i from 0 to 255 do
+///     fi := sum(j = 0 to l−1)(βil+j 2j)
+/// end for
+/// return f0 +f1X +f2X2 +···+f255X255
+/// ```
+///
+/// The Kyber Round 3 specification can be found at:
+/// https://pq-crystals.org/kyber/data/kyber-specification-round3-20210131.pdf
+///
+pub(crate) fn deserialize_ring_element(ring_element_bytes : [u8; parameters::BITS_PER_RING_ELEMENT / 8]) -> RingElement {
```

Comment:
This is unused. I presume it's going to be used?

Reply:
Yes, it will be used during Decapsulation.